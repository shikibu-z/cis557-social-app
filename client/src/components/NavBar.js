/* eslint-disable operator-linebreak */
/* eslint-disable import/no-extraneous-dependencies */
import { React } from 'react';
import {
  Navbar, Container, Nav, Button,
} from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../stylesheets/NavBar.css';
import {
  HouseDoorFill, CollectionFill, ChatFill, PersonFill, BoxArrowInRight,
} from 'react-bootstrap-icons';
import { useHistory } from 'react-router';

function NavBar(props) {
  const { isLogin } = props;
  const history = useHistory();
  const handleLogout = () => {
    // cleanup user info when logging out
    localStorage.clear();
    history.push('/login');
  };
  return (
    <div>
      <Navbar className="menu" bg="dark" variant="dark">
        <Container fluid>
          <Navbar.Brand href="#home">
            <img
              id="logo"
              src="../Connexion-logos_transparent.png"
              alt="connexion"
              className="d-inline-block align-top"
            />{' '}
          </Navbar.Brand>
          {isLogin === 0 &&
            (
              <div>
                <Navbar.Toggle aria-controls="navbarScroll" />
                <Navbar.Collapse id="navbarScroll">
                  <Nav
                    className="me-auto my-2 my-lg-0"
                    style={{ maxHeight: '100px' }}
                    navbarScroll
                  >
                    <Button id="home" className="NavBar-buttons" onClick={() => { history.push('/homepage'); }}><HouseDoorFill /></Button>
                    <Button id="createGroup" className="NavBar-buttons" onClick={() => { history.push('/explore'); }}><CollectionFill /></Button>
                    <Button id="chat" className="NavBar-buttons" onClick={() => { history.push('/chat'); }}><ChatFill /></Button>
                    <Button id="profile" className="NavBar-buttons" onClick={() => { history.push('/personalprofile'); }}><PersonFill /></Button>
                    <Button id="logout" className="NavBar-buttons" onClick={handleLogout}><BoxArrowInRight /></Button>
                  </Nav>
                </Navbar.Collapse>
              </div>
            )}
        </Container>
      </Navbar>
    </div>
  );
}

export default NavBar;
