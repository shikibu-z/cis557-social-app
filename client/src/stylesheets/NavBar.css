.menu {
    background-color: #F8B114 !important
}

#logo {
    height: auto;
    width: 8%;
}

.NavBar-buttons {
    background-color: white !important;
    border-color: #F8B114;
    border-radius: 20px;
    color: black;
    margin-left: 2%;
    margin-right: 2%;
    width: 60px !important;
    height: 60px !important;
}

.NavBar-buttons:hover {
    background-color: black !important;
    color: white;
    border-color: black;
}

.me-auto.my-2.my-lg-0 {
    position: relative;
    margin-left: -20% !important;
}