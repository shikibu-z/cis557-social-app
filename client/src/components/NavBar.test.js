/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';

test('Navbar component matches snapshot', () => {
  const component = renderer.create(<NavBar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
