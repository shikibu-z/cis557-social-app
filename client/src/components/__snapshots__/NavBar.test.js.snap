// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches snapshot 1`] = `
<div>
  <nav
    className="menu navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#home"
      >
        <img
          alt="connexion"
          className="d-inline-block align-top"
          id="logo"
          src="../Connexion-logos_transparent.png"
        />
         
      </a>
    </div>
  </nav>
</div>
`;
